src/App/Controller/PrivateDiscussionController.php

```diff
@@ -0,0 +1,325 @@
+<?php
+
+namespace App\Controller;
+
+use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+use Symfony\Component\Security\Core\Exception\AccessDeniedException;
+use Symfony\Component\HttpFoundation\Request;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Security;
+use Symfony\Component\Security\Core\User\UserInterface;
+use App\Entity\Post;
+use App\Entity\PrivateDiscussion;
+use App\Form\Post\EditPostType;
+use App\Form\Discussion\AddPrivateDiscussionType;
+use App\Form\Discussion\EditDiscussionType;
+use App\Form\SelectUserType;
+use App\Form\Discussion\ChangeAdminType;
+
+class PrivateDiscussionController extends Controller
+{
+    /**
+     * @Security("has_role('ROLE_USER')")
+     */
+    public function showAction(Request $request, $slug)
+    {
+        $discussion = $this->getDoctrine()->getRepository('App:PrivateDiscussion')->findOneBySlug($slug);
+
+        if (null === $discussion) {
+            throw $this->createNotFoundException();
+        }
+
+        if (false === $this->get('security.context')->isGranted('view', $discussion)) {
+            throw new AccessDeniedException();
+        } else {
+            $discussion->removeUnreader($this->getUser());
+            $em = $this->getDoctrine()->getManager();
+            $em->persist($discussion);
+            $em->flush();
+        }
+
+        if (false === $discussion->isLocked()) {
```

Comment:
I would do the opposite (`if ($discussion->isLocked())`). If it's locked, you just return the template with the discussion. Think "early return". It makes your code easy to read and follow :).